src/components/DepositWidget/index.tsx

```diff
@@ -72,6 +99,152 @@ const DepositWidget: React.FC = () => {
     <EtherscanLink type="contract" identifier={contractAddress} label={<small>View verified contract</small>} />
   )
 
+  function _getToken(tokenAddress: string): TokenBalanceDetails {
+    return balances.find(({ address }) => address === tokenAddress)
+  }
+
+  function _updateToken(tokenAddress: string, updateBalances: Mutation<TokenBalanceDetails>): void {
+    setBalances(balances =>
+      balances.map(tokenBalancesAux => {
+        const { address: tokenAddressAux } = tokenBalancesAux
```

Comment:
This is getting complicated for `setState` . At this point I would `useReducer`, though not strictly necessary, with actions for `highlight | deposit | claim | withdraw`

Reply:
I thought the same, this part was coded on the plane on my way to Devcon, and that same trip I made a simulation of using a reducer here, and I didn't like where it was heading so I didn't continue that path. 

I think it's not that horrible for now. My attempt to use the reducer was in a branch called https://github.com/gnosis/dex-react/tree/use-reducer

Reply:
Although is not related, I think this PR #142 simplifies the widget, so it's not so strictly necessary the reducer imo.